post_processing/ci_plots.jl

```diff
@@ -280,13 +280,58 @@ function make_plots_generic(
 end
 
 """
-    make_spectra_generic
+    compute_spectrum(var)
 
-Use ClimaCoreSpectra to compute and plot spectra for the given `vars`.
+Compute the spectrum associated to the given variable. Returns a ClimaAnalysis.OutputVar.
+"""
+function compute_spectrum(var)
+    # power_spectrum_2d seems to work only when the two dimensions have precisely one
+    # twice as many points as the other
+    dim1, dim2, dim3 = var.index2dim[1:3]
 
-Extra arguments are passed to `ClimaAnalysis.slice`
+    len1 = length(var.dims[dim1])
+    len2 = length(var.dims[dim2])
+
+    len1 == 2len2 || error("Cannot take this spectrum ($len1 != 2 $len2)")
+
+    (dim1 == "lon" || dim1 == "long") ||
+        error("First dimension has to be longitude (found $dim1)")
+    dim2 == "lat" || error("Second dimension has to be latitude (found $dim2)")
+    dim3 == "z" || error("Third dimension has to be altitude (found $dim3)")
+
+    FT = eltype(var.data)
+    mass_weight = ones(FT, length(var.dims[dim3]))
```

Comment:
We're hardcoding weights here; could this be passed as an argument to this function with this default value instead?